examples/erc1155-supply/tests/erc1155-supply.rs

```diff
@@ -0,0 +1,401 @@
+#![cfg(feature = "e2e")]
+
+use abi::Erc1155Supply;
+use alloy::{
+    primitives::{Address, U256},
+    rpc::types::TransactionReceipt,
+};
+use e2e::{
+    receipt, send, watch, Account, EventExt, Panic, PanicCode, ReceiptExt,
+};
+
+mod abi;
+
+fn random_token_ids(size: usize) -> Vec<U256> {
+    (0..size).map(U256::from).collect()
+}
+
+fn random_values(size: usize) -> Vec<U256> {
+    (0..size).map(|_| U256::from(rand::random::<u128>())).collect()
+}
+
+async fn deploy_and_mint(
+    receiver: &Account,
+    size: usize,
+) -> eyre::Result<(Address, Vec<U256>, Vec<U256>, TransactionReceipt)> {
+    let contract_addr = receiver.as_deployer().deploy().await?.address()?;
+    let contract = Erc1155Supply::new(contract_addr, &receiver.wallet);
+
+    let token_ids = random_token_ids(size);
+    let values = random_values(size);
+
+    let receipt = receipt!(contract.mintBatch(
+        receiver.address(),
+        token_ids.clone(),
+        values.clone(),
+        vec![].into()
+    ))?;
+
+    Ok((contract_addr, token_ids, values, receipt))
+}
+
+// ============================================================================
+// Integration Tests: ERC-1155 Supply Extension
```

Comment:
IMHO you should copy E2E tests for happy paths of re-exported fns to make sure that everything works as expected.

Reply:
Sounds reasonable